### PR TITLE
Added 6 python algorithms

### PR DESCRIPTION
**Pull request by jingwu74**

### CODE REVIEW

File: Algorithm.Python/EmaCrossUniverseSelectionAlgorithm.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
+﻿from System.Collections.Generic import List
```
Comment (jaredbroad) [inline Algorithm.Python/EmaCrossUniverseSelectionAlgorithm.py]: Please add the apache license header

File: Algorithm.Python/BasicTemplateForexAlgorithm.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        history = self.History(TimeSpan.FromSeconds(5), Resolution.Second)
+         
+        for data in sorted(history, key=lambda x: x.Time):
+            for key in data.Keys:
+                self.Log(str(key.Value) + ": " + str(data.Time) + " > " + str(data[key].Value))
+        
+    def OnData(self, data):
+        # Print to console to verify that data is coming in
+        for key in data.Keys:
+            self.Log(str(key.Value) + ": " + str(data.Time) + " > " + str(i[key].Value))
```
Comment (Martin-Molinero) [inline Algorithm.Python/BasicTemplateForexAlgorithm.py]: is there a typo in the last str?

File: Algorithm.Python/BasicTemplateForexAlgorithm.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        self.AddForex("GBPUSD", Resolution.Minute)
+        self.AddForex("EURGBP", Resolution.Minute)
+        
+        self.History(5, Resolution.Daily)
+        self.History(5, Resolution.Hour)
+        self.History(5, Resolution.Minute)
+
+        history = self.History(TimeSpan.FromSeconds(5), Resolution.Second)
+         
+        for data in sorted(history, key=lambda x: x.Time):
```
Comment (Martin-Molinero) [inline Algorithm.Python/BasicTemplateForexAlgorithm.py]: data is not sorted in history?

File: Algorithm.Python/EmaCrossUniverseSelectionAlgorithm.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        # Filter the values of the dict: we only want up-trending securities 
+        values = filter(lambda x: x.is_uptrend, self.averages.values())
+        
+        # Sorts the values of the dict: we want those with greater difference between the moving averages
+        values.sort(key=lambda x: x.scale, reverse=True)
+
+        # we need to return only the symbol objects
+        list = List[Symbol]()
+        for x in values[:self.coarse_count]:
+            print x.symbol, x.scale
```
Comment (Martin-Molinero) [inline Algorithm.Python/EmaCrossUniverseSelectionAlgorithm.py]: Hi ! I see you generally use self.Log() & here you used print, should it be self.Log() too?

File: Algorithm.Python/BasicTemplateForexAlgorithm.py (hunk 3 above)
Comment (jingwu74) [reply Algorithm.Python/BasicTemplateForexAlgorithm.py]: It is sorted in history

File: Algorithm.Python/EmaCrossUniverseSelectionAlgorithm.py (hunk 4 above)
Comment (jingwu74) [reply Algorithm.Python/EmaCrossUniverseSelectionAlgorithm.py]: Yes, it should be Log

**Pull request merged**